### PR TITLE
HTTP 403 for invalid login

### PR DESCRIPTION
**Pull request by Alkarex**
https://github.com/FreshRSS/FreshRSS/issues/1015
And does not leak if user exists or not.
Must not use redirect to send status 403.

### CODE REVIEW

File: app/Controllers/authController.php [hunk 1]
```diff
@@ -123,8 +123,7 @@ public function formLoginAction() {
 
 			$conf = get_user_configuration($username);
 			if (is_null($conf)) {
-				Minz_Request::bad(_t('feedback.auth.login.invalid'),
-				                  array('c' => 'auth', 'a' => 'login'));
+				Minz_Error::error(403, array(_t('feedback.auth.login.invalid')), false);
```
Comment (marienfressinaud) [inline app/Controllers/authController.php:126]: If there is no redirection (`$redirect = false`), rest of the code in the method is executed as well. Are you sure it does not cause more troubles?

Comment (Alkarex) [inline app/Controllers/authController.php:126]: Ok, we need to return/exit after this line, then.

Comment (Alkarex) [inline app/Controllers/authController.php:126]: This one was problematic indeed. I have added a `return`.
The other two below should be fine.

File: app/Controllers/javascriptController.php [hunk 2]
```diff
@@ -43,7 +43,11 @@ public function nonceAction() {
 		} else {
 			Minz_Log::notice('Nonce failure due to invalid username!');
 		}
-		$this->view->nonce = '';	//Failure
-		$this->view->salt1 = '';
+		//Failure: Return random data.
+		$this->view->salt1 = sprintf('$2a$%02d$', FreshRSS_user_Controller::BCRYPT_COST);
+		for ($i = 22; $i > 0; $i--) {
+			$this->view->salt1 .= './ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'[rand(0, 63)];
```
Comment (marienfressinaud) [inline app/Controllers/javascriptController.php]: Just realized: is this syntax valid with PHP 5.3?

Comment (Alkarex) [inline app/Controllers/javascriptController.php]: Well spotted. It is indeed first PHP 5.5 that added support for accessing characters within string literals using []...
http://php.net/manual/en/language.types.string.php

**Pull request merged**